tests/journald_tests.cpp

```diff
@@ -302,6 +303,187 @@ TEST_F(JournaldLoggerTest, ROOT_LogToJournaldWithBigLabel)
 }
 
 
+// Loads the journald ContainerLogger module and checks for the
+// non-existence of the logrotate config file.
+TEST_F(JournaldLoggerTest, ROOT_LogrotateCustomOptions)
+{
+  const std::string testFile = path::join(sandbox.get(), "CustomRotateOptions");
+
+  // Custom config consists of a postrotate script which creates
+  // an empty file in the temporary directory on log rotation.
+  const std::string customConfig =
+    "postrotate\n touch " + testFile + "\nendscript";
+
+  // There is no way to change a task's custom logrotate options except when
+  // loading the module.  So this test will unload the module, change the
+  // logrotate options, and then reload the module.
```

Comment:
I hesitated to create a new `modules.json.in` file (or modify the existing one) because there is currently only one test that needs to supply custom logrotate options.  If you look at the commits individually, you'll see that the third commit adds the test using an environment variable to modify the custom options.  And the fourth commit needs to do this unload-reload operation, because it disables the environment variable.

In addition, for better cleanup, the custom option touches a file in the sandbox, which won't be known until runtime.  I considered the following alternatives:
* Making a new test fixture for loading a modified module config.  This would result in even more duplication, and doesn't seem worthwhile for a single test.
* Adding the custom config to all tests, with a fixed `touch /path/to/temp/testFile` clause.  This doesn't cause other tests to fail (because they don't check for rotation in the same way).  But it feels undesirable to mix test requirements up like this.

The unload-reload at least reuses the JSON parsing logic from the test fixture 😄